.github/workflows/ci_test.yml

```
@@ -17,7 +17,7 @@ jobs:
   ci-tests:
     name: ${{ matrix.os }}, ${{ matrix.tox_env }}
     runs-on: ${{ matrix.os }}
-    if: "!(contains(github.event.head_commit.message, '[skip ci]') || contains(github.event.head_commit.message, '[ci skip]'))"
+    if: ${{ !contains(github.event.head_commit.message, '[docs only]') }}
```

Try:
```suggestion
    if: ! contains(github.event.head_commit.message, '[docs only]')
```

When using expressions in an `if`, we should omit the `${{ }}`, otherwise it turns the result into a string which is always true ([source](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif)).

I think the `[skip ci]` stuff worked not because of this line, but because it's hard-coded in GitHub itself ([source](https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/)).